### PR TITLE
Fix: Attachment file issue

### PR DESCRIPTION
**Pull request by RaiVaibhav**
Previously during uploading attachment the whole analysis component get's update which which reseting the input component value to the value save in backend.

### CODE REVIEW

File: app/components/edit-analysis-details.js [hunk 1]
```diff
@@ -127,18 +128,85 @@ export default Component.extend({
...
       const fileName = file.blob.name;
       const data = {
         name: fileName
       };
       const analysisid= this.get("analysis.analysisid");
       try {
-        var fileData = await this.get("ajax").post(ENV.endpoints.uploadFile,{namespace: 'api/hudson-api', data});
-        await file.uploadBinary(fileData.url, {
+        var fileData = yield this.get("ajax").post(ENV.endpoints.uploadFile,{namespace: 'api/hudson-api', data});
+        file.uploadBinary(fileData.url, {
```
Comment (cosmosgenius) [inline app/components/edit-analysis-details.js]: this needs to be yielded

File: app/components/edit-analysis-details.js [hunk 2]
```diff
@@ -149,19 +217,27 @@ export default Component.extend({
...
         };
-        await this.get("ajax").post(ENV.endpoints.uploadedAttachment,{namespace: 'api/hudson-api', data: fileDetailsData});
-
-        this.set("isUploading", false);
-        this.get("notify").success("File Uploaded Successfully");
-        const analysisObj = this.get("store").findRecord('security/analysis', this.get("analysis.analysisid"));
-        this.set('analysisDetails', analysisObj);
+        yield this.get("ajax").post(ENV.endpoints.uploadedAttachment,{namespace: 'api/hudson-api', data: fileDetailsData});
+        yield this.get('detailSaveAjaxCallUtil').perform()
+        .then(() => {
```
Comment (cosmosgenius) [inline app/components/edit-analysis-details.js]: Don't use then here use the normal code to continue execution

File: app/templates/components/edit-analysis-details.emblem [hunk 3]
```diff
@@ -191,9 +191,9 @@
         .column
           form
             p.control.is-fullwidth
-              = file-upload name="files" accept="*/*" onfileadd=(action "uploadFile") as |queue|
-                a class="button is-primary" disabled=isUploading
-                  if isUploading
+              = file-upload name="files" accept="*/*" onfileadd=( action 'uploadFileWrapper' ) as |queue0|
```
Comment (cosmosgenius) [inline app/templates/components/edit-analysis-details.emblem]: queue0?

File: app/components/edit-analysis-details.js [hunk 4]
```diff
@@ -127,18 +128,85 @@ export default Component.extend({
...
   },
 
-  actions: {
 
-    async uploadFile(file) {
+
+  detailSaveUtil: task(function * (param){
+    let key=param;
+    yield this.get('detailSaveAjaxCallUtil').perform(param)
+    .then(() => {
```
Comment (cosmosgenius) [inline app/components/edit-analysis-details.js]: Don't use then there as well use normal code. You can use normal try catch to catch errors.

File: app/templates/components/edit-analysis-details.emblem (hunk 3 above)
Comment (RaiVaibhav) [reply app/templates/components/edit-analysis-details.emblem]: @cosmosgenius  will fix all :+1: , this one I think gets added by mistake during committing the changes, thanks for the review  :+1:

File: app/components/edit-analysis-details.js [hunk 5]
```diff
@@ -127,18 +128,85 @@ export default Component.extend({
...
-  actions: {
 
-    async uploadFile(file) {
+
+  detailSaveUtil: task(function * (param){
+    let key=param;
+    try{
+      yield this.get('detailSaveAjaxCallUtil').perform(param);
+      this.get("notify").success("Analyses Updated");
+      if(key) {
```
Comment (Praseetha-KR) [inline app/components/edit-analysis-details.js]: Better to do `key == 'back'` check

**Pull request merged**